### PR TITLE
UI event tests for focus and mouse-related events

### PR DESCRIPTION
**Pull request by sunyunjia**

### CODE REVIEW

File: uievents/order-of-events/focus-events/focus-manual.html [hunk 1]
```diff
@@ -1,81 +1,76 @@
...
+  EventRecorder.addEventListenersForNodes(relevantEvents, inputs, stopPropagation);
+  var expected = [
+  {type: "focusin", target: "a"},
+  {type: "focus", target: "a"},
+  {type: "focusout", target: "a"},
+  {type: "focusin", target: "b"},
+  {type: "blur", target: "a"},
+  {type: "focus", target: "b"},
+  {type: "focusout", target: "b"},
+  {type: "focusin", target: "a"},
```
Comment (dtapuska) [inline uievents/order-of-events/focus-events/focus-manual.html]: This focusin seems odd.

**Review by dtapuska — APPROVED** [review]
lgtm

File: uievents/order-of-events/focus-events/focus-contained-manual.html [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+      mergeEventTypes: ["mousemove"],
+      objectMap: {
+         "a": a,
+         "b": b,
+      }
+    });
+
+  EventRecorder.addEventListenersForNodes(relevantEvents, inputs, stopPropagation);
+  var expected = [
+  {type: "focusin", target: "b"},
```
Comment (garykac) [inline uievents/order-of-events/focus-events/focus-contained-manual.html]: The indent style here is odd and doesn't match the rest of the file. Why aren't these indented 2 spaces with the closing bracket on its own line?

File: uievents/order-of-events/mouse-events/click-order-manual.html [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    {type: "mousemove", target: "blue", optional: true},
+    {type: "mouseup", target: "blue"},
+    {type: "click", target: "blue"},
+    {type: "mousemove", target: "blue"},
+    {type: "mousedown", target: "blue"},
+    {type: "mousemove", target: "blue", optional: true},
+    {type: "mouseup", target: "blue"},
+    {type: "click", target: "blue"},
+    {type: "dblclick", target: "blue"},
+    {type: "mousemove", target: "blue", optional: true}];
```
Comment (garykac) [inline uievents/order-of-events/mouse-events/click-order-manual.html]: This block is indented correctly, but the closing ]; is not consistent with the rest of the file.

File: uievents/order-of-events/mouse-events/mousemove-across-manual.html [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+  {type: "mouseenter", target: "b1"},
+  {type: "mouseenter", target: "a1"},
+  {type: "mousemove", target: "a1", optional: true},
+  {type: "mouseout", target: "a1"},
+  {type: "mouseleave", target: "a1"},
+  {type: "mouseleave", target: "b1"},
+  {type: "mouseover", target: "c1"},
+  {type: "mousemove", target: "c1", optional: true},
+  {type: "mouseout", target: "c1"},
+  {type: "mouseleave", target: "c1"},
```
Comment (garykac) [inline uievents/order-of-events/mouse-events/mousemove-across-manual.html]: Indent

File: uievents/order-of-events/mouse-events/mousemove-across-manual.html [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+  width: 200px;
+  background: black;
+}
+</style>
+<p>
+  Steps:
+  <ol>
+    <li>Move your mouse across the yellow/red <code>&lt;div&gt;</code> element quickly from right to left.
+    <li>Move your mouse across the black/blue <code>&lt;div&gt;</code> element quickly from right to left.
+    <li>If the test fails, redo it again and move faster on the black/blue <code>&lt;div&gt;</code> element next time.
```
Comment (garykac) [inline uievents/order-of-events/mouse-events/mousemove-across-manual.html]: Why does moving faster make the test more likely to succeed? If it's just looking at order of event firing, then this should not be relevant.

Comment (sunyunjia) [inline uievents/order-of-events/mouse-events/mousemove-across-manual.html]: I added a comment in the UI Event Test doc referring to this.

**Comment by dtapuska** [comment]
@garykac  Gary you good with this now?

**Comment by garykac** [comment]
yep, lgtm

**Comment by dtapuska** [comment]
@RByers please merge

**Pull request merged**